lib/vector-core/src/mapping/parser/mod.rs

```diff
@@ -214,6 +214,7 @@ fn query_function_from_pairs(mut pairs: Pairs<Rule>) -> Result<Box<dyn query::Fu
     signature.into_boxed_function(arguments)
 }
 
+#[allow(clippy::redundant_closure_for_method_calls)]
```

Comment:
What was this one complaining about here / why couldn't we resolve it?

Reply:
I had mistakenly believed I couldn't fix it. Resolved in 29780a0523b511a604a58641638c7d2b550c7dee